### PR TITLE
Enforce nLockTime/sequence transaction finality at submit

### PR DESCRIPTION
**Pull request by sirdeggen**
## Summary
- Transaction validation previously had no finality check — `spv.Verify` covers only fees, merkle paths, and script execution.
- Adds a finality gate to `ValidateTransaction` mirroring teranode `util.IsTransactionFinal` (consensus rule TNJ-13).

## Changes
- `validator/finality.go` (new): `IsFinal` + `validLockTime` mirror teranode `util.IsTransactionFinal` / `util.ValidLockTime`. A tx is final when every input sequence is `0xffffffff`, `nLockTime == 0`, or `nLockTime` is satisfied (`< 500000000` → block height; `>=` → timestamp/MTP). `checkFinality` sources height-based locks from the chain tip (next-block height) and time-based locks from the wall clock; chain-tip read failures accept rather than block submission.
- `validator/validator.go`: call `checkFinality` in `ValidateTransaction` between policy and `spv.Verify`; map `ErrTxNotFinal` → `StatusMalformed`. Added nil-safe `chainTip`/`now` fields with `SetChainTip`/`SetNowFunc`.
- `app/app.go`: wire the store (`GetActiveTipBlockHeight`) as the chain-tip source.

## Test plan
- [x] `go test ./validator/` — 14 finality tests pass (locktime-zero, all-inputs-final, height/time pass+fail, mixed sequences, threshold edge cases either side of 500000000, nil-tip skip, tip-error accept)
- [x] `go build ./...`, `go vet ./validator/ ./app/` clean

Closes ARC issue: https://github.com/bitcoin-sv/arc/issues/145

### CODE REVIEW

File: validator/finality.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+		return ErrTxNotFinal
+	}
+
+	// Time-based locktime: wall clock is >= median-time-past, so a lock that
+	// has elapsed against now() has certainly elapsed against MTP.
+	now := v.now
+	if now == nil {
+		now = func() int64 { return time.Now().Unix() }
+	}
+	if validLockTime(tx.LockTime, 0, clampI64ToU32(now())) {
```
[the remark this answers is not in the document]
Comment (galt-tr) [reply validator/finality.go:109]: @sirdeggen thoughts?

[the remark this answers is not in the document]
Comment (sirdeggen) [reply validator/finality.go:109]: Yeah this definitely needs to be fixed, the MTP can be calculated with header data only I believe that's possible via chaintracks.

**Comment by galt-tr** [comment]
Thanks for this — it correctly identified the missing finality check, and #246 builds on the same teranode `util.IsTransactionFinal` framing. While diagnosing #245 we found two gaps here that #246 addresses, so proposing we close this PR in favor of that one:

1. **Wall clock vs MTP.** This PR gates time-based locktimes on `time.Now()`, reasoning that a lock elapsed against now() has certainly elapsed against MTP. The implication actually runs the other way (wall clock ≥ MTP means MTP-final ⇒ now-final, not the reverse). The #245 transaction is the live counterexample: locktime 21:41:50 UTC, submitted 23:20–23:41 (passes a wall-clock gate) while mainnet MTP was 20:49–21:31 — teranode rejected it with `UTXO_NON_FINAL (71) → TX_LOCK_TIME (35): lock time (1783806110) as timestamp is not less than median block time (1783805456)` (from node logs; the client response strips this). The common wallet failure mode — anti-fee-sniping locktime ≈ signing time, submitted within MTP's 1–2.5h lag — sits precisely in the window a wall-clock gate can't see. #246 compares against real MTP (median of the last 11 headers via chaintracks, embedded or remote).

2. **Boundary inequality.** `validLockTime` here uses `>=` where teranode/bitcoin-sv use strict `>`, so a tx with `lockTime == tip+1` passes this gate and still bounces off teranode generically. #246 mirrors teranode's strict inequalities with tests pinning both boundaries.

#246 also carries the fail-open-on-chain-state-failure behavior from this PR, plus an actionable rejection message (locktime, current MTP, ETA, remediation), metrics, and an end-to-end smoke test.
